### PR TITLE
Add --max-results flag to range list command

### PR DESCRIPTION
**Pull request by thundercw**
1. Add --max-results flag to range list command. 
2. Add a log to display the total results count.

### CODE REVIEW

File: cli/kv.go [hunk 1]
```diff
@@ -318,6 +314,7 @@ func showResult(rows []client.KeyValue) {
 		key := proto.Key(row.Key)
 		fmt.Printf("%s\t%q\n", key, row.Value)
 	}
+	fmt.Printf("Total got: %d\n", len(rows))
```
Comment (petermattis) [inline cli/kv.go]: This reads a bit awkward. It is more lines of code, but how about:

```
  suffix := "s"
  if len(rows) == 1 {
    suffix = ""
  }
  fmt.Printf("%d result%s\n", len(rows))
```

**Comment by petermattis** [comment]
LGTM after a small tweak to the log message.

File: cli/kv.go (hunk 1 above)
Comment (thundercw) [inline cli/kv.go]: Sounds good.

**Comment by thundercw** [comment]
@petermattis Updated . PTAL

File: cli/range.go [hunk 2]
```diff
@@ -74,6 +72,7 @@ func runLsRanges(cmd *cobra.Command, args []string) {
 				i, replica.NodeID, replica.StoreID)
 		}
 	}
+	fmt.Printf("%d results\n", len(rows))
```
Comment (petermattis) [inline cli/range.go]: I used 4 lines of code to get the proper pluralization of results. Irritating, yes, but `1 results` looks odd and wrong to me.

Comment (thundercw) [inline cli/range.go]: How about  `result(s)` ?

Comment (petermattis) [inline cli/range.go]: Sure, that works for me.

File: cli/flags.go [hunk 3]
```diff
@@ -226,10 +228,10 @@ func initFlags(ctx *server.Context) {
 		f.StringVar(&ctx.Certs, "certs", ctx.Certs, flagUsage["certs"])
 	}
 
-	// Max results flag for scan and reverse scan.
-	for _, cmd := range []*cobra.Command{scanCmd, reverseScanCmd} {
+	// Max results flag for scan, reverse scan and range list.
```
Comment (tamird) [inline cli/flags.go]: s/and/, and/

**Comment by thundercw** [comment]
@petermattis  @tamird  updated. PTAL.

**Comment by petermattis** [comment]
LGTM

**Comment by tamird** [comment]
LGTM

**Comment by thundercw** [comment]
I will merge it after all checks are complete.

**Comment by tamird** [comment]
@thundercw i'm going to combine a few cli-related PRs, let me pull this one in.

**Comment by thundercw** [comment]
ok

**Pull request closed without merging**